src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -551,16 +573,27 @@ public void validateForStudentExamGeneration(Exam exam) throws BadRequestAlertEx
                     exam.addRegisteredUser(student);
                     continue;
                 }
+
                 // 2) if we cannot find the student, we use the registration number and try to find the student in the (TUM) LDAP, create it in the Artemis DB and in a potential
                 // external user management system
                 optionalStudent = userService.createUserFromLdap(registrationNumber);
+                if (optionalStudent.isEmpty()) {
+                    // An invalid registration number might be corrected by getting the number from the login
+                    var userOptional = userService.getUserByLogin(studentDto.getLogin());
+                    if (userOptional.isPresent()) {
+                        registrationNumber = userOptional.get().getRegistrationNumber();
+                        optionalStudent = userService.createUserFromLdap(registrationNumber);
+                    }
+                }
```

Comment:
I do not understand this. We did not find the user locally, we also did not get the user from ldap, now check again locally? Did we not check this unsuccessfully before?

Reply:
Good catch, I must not overwrite `registrationNumber` here and above.